state/database.go

```diff
@@ -102,6 +102,10 @@ type Database interface {
 	// Schema returns the schema used to load the database. The returned schema
 	// is not a copy and must not be modified.
 	Schema() CollectionSchema
+
+	// NoChangeLog returns the same Database without changelog recording.
+	// NOTE: Only to be used by txns.log upgrade from client-side txn to server-side txn.
+	NoChangeLog() Database
```

Comment:
Can we not put this here but instead create a NoChangeLogDatabase interface (or whatever) which embeds Database and adds this method. And then use only in the upgrade.

Reply:
Made a wrapper function instead that does this rather than polluting the Database interface.